src/StatusQ/Controls/StatusTokenInlineSelector.qml

```diff
@@ -0,0 +1,170 @@
+import QtQuick 2.14
+import QtQuick.Layouts 1.14
+
+import StatusQ.Core 0.1
+import StatusQ.Components 0.1
+import StatusQ.Core.Theme 0.1
+
+/*!
+   \qmltype StatusTokenInlineSelector
+   \inherits RowLayout
+   \inqmlmodule StatusQ.Controls
+   \since StatusQ.Controls 0.1
+   \brief It presents selectable tokens in inline form.
+
+   The \c StatusTokenInlineSelector is a graphical component that is meant to be used as inline text input.
+
+   Example of how the component looks like:
+   \image status_token_inline_selector.png
+*/
+
+RowLayout {
+    id: root
+
+    /*!
+       \qmlproperty var StatusTokenInlineSelector::tokens
+       This is a REQUIRED property that contains an array of objects that describes each token.
+
+       \qml
+            tokens: [{amount: 0.1, token: "ETH"}, {amount: 10, token: "SNT"}]
+       \endqml
+    */
+    property var tokens
+
+    /*!
+       \qmlproperty var StatusTokenInlineSelector::tokenImageSourceGetter
+       This is a property function used to acquire image source for given token.
+    */
+    property var tokenImageSourceGetter: function (token) {
+        return "../../assets/img/icons/%1.png".arg(token)
+    }
+
+    signal triggered(real amount, string token)
+
+    StatusBaseText {
+        text: qsTr("Hold")
+        color: Theme.palette.directColor6
+        font.pixelSize: 15
+    }
+
+    Repeater {
+        model: Math.max(0, root.tokens.length - 1)
+        delegate: Loader {
+            sourceComponent: tokenComponent
+            onLoaded: d.assignItemProperties(item, index, root.tokens[index])
+        }
+    }
+
+    StatusBaseText {
+        text: qsTr("or")
+        color: Theme.palette.directColor6
+        font.pixelSize: 15
+    }
+
+    Loader {
+        active: root.tokens.length > 1
+        sourceComponent: tokenComponent
+        onLoaded: {
+            const index = root.tokens.length - 1
+            d.assignItemProperties(item, index, root.tokens[index])
+        }
+    }
+
+    StatusBaseText {
+        text: qsTr("to post")
+        color: Theme.palette.directColor6
+        font.pixelSize: 15
+    }
+
+    QtObject {
```

Comment:
Just a question here, just to agree with coding conventions... I use to create that `QtObject` which is used to create "private" things here (this is qt's proposed order, so I write it after the functions definition, since normally this object contains properties and functions that will be used in the following lines and they are also properties and functions, but we can't expose them as an interface):

- id
- property declarations
- signal declarations
- JavaScript functions
- -->> PRIVATE STUFF DEFINITION <<--
- object properties
- child objects
- states
- transitions 

WDYT @osmaczko ?

Reply:
Yeah, it's a matter of agreement on coding conventions, which we need to define, see: https://github.com/status-im/StatusQ/pull/594#discussion_r832579471

I don't have a strong preference, I tend to put private stuff/details at the end.

Reply:
> Yeah, it's a matter of agreement on coding conventions, that we need to define, see: [#594 (comment)](https://github.com/status-im/StatusQ/pull/594#discussion_r832579471)
> 
> I don't have a strong preference, I tend to put private stuff/details at the end.
Ok! I will check it there.. still not reviewed!!